tests/autologging/test_autologging_utils.py

```diff
@@ -642,3 +644,134 @@ def test_autologging_event_logger_default_impl_warns_for_log_autolog_called_with
             call_args=("a"),
             call_kwargs={"b": "c"},
         )
+
+
+def test_is_autologging_supported():
+    assert _check_version_in_range("1.0.2", "1.0.1", "1.0.3")
+    assert _check_version_in_range("1.0.1", "1.0.1", "1.0.3")
+    assert _check_version_in_range("1.0.3", "1.0.1", "1.0.3")
+    assert not _check_version_in_range("1.0.0", "1.0.1", "1.0.3")
+    assert not _check_version_in_range("1.0.4", "1.0.1", "1.0.3")
+    assert not _check_version_in_range("0.99.99", "1.0.1", "1.0.3")
+    assert not _check_version_in_range("1.1.0", "1.0.1", "1.0.3")
+    assert _check_version_in_range("1.0.3", "1.0.1", "1.0.3.post1")
+
+    def gen_get_module_version_fn(module_name_to_version_map):
+        def get_model_version_fn(module_name):
+            if module_name in module_name_to_version_map:
+                return module_name_to_version_map[module_name]
+            raise ValueError("Test error: unknown module name: " + module_name)
+
+        return get_model_version_fn
+
+    assert _is_autologging_integration_supported(
+        "fastai", gen_get_module_version_fn({"fastai": "1.0.60"})
+    )
+    assert not _is_autologging_integration_supported(
+        "fastai", gen_get_module_version_fn({"fastai": "1.0.50"})
+    )
```

Comment:
What happens if we make the following change to `ml-package-versions.yml`?

```diff
fastai-1.x:
  package_info:
    pip_release: "fastai"

  models:
-   minimum: "1.0.60"
+   minimum: "1.0.30"
```

Reply:
Can we mock `_version_file_json` and do something like below?

```python
with mock.patch.object("_version_file_json", new_value={"min": "0.24.0", "max": "0.25.0"}):
    with mock.patch("sklearn.__version__", new_value="0.23.0"):
        assert not is_autologging_supported("sklearn")

    ...
```

@dbczumar Any thoughts on this?

Reply:
Updated. Mock _version_file_json dict.